scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/PPMLContextMultiKMS.scala

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.ppml
+
+import com.intel.analytics.bigdl.dllib.NNContext.{checkScalaVersion, checkSparkVersion, createSparkConf, initConf, initNNContext}
+import com.intel.analytics.bigdl.dllib.utils.Log4Error
+import com.intel.analytics.bigdl.ppml.crypto.{AES_CBC_PKCS5PADDING, BigDLEncrypt, Crypto, CryptoMode, DECRYPT, ENCRYPT, EncryptRuntimeException, PLAIN_TEXT}
+import com.intel.analytics.bigdl.ppml.utils.Supportive
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.spark.input.PortableDataStream
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, DataFrameReader, DataFrameWriter, Row, SparkSession}
+import com.intel.analytics.bigdl.ppml.kms.{AzureKeyManagementService, EHSMKeyManagementService, KMS_CONVENTION,
+KeyManagementService, SimpleKeyManagementService}
+import com.intel.analytics.bigdl.ppml.crypto.dataframe.{EncryptedDataFrameReader, EncryptedDataFrameWriter}
+import org.apache.hadoop.fs.Path
+
+/**
+ * PPMLContext who wraps a SparkSession and provides read functions to
+ * read encrypted data files to plain-text RDD or DataFrame, also provides
+ * write functions to save DataFrame to encrypted data files.
+ * @param kms
+ * @param sparkSession
+ */
+class PPMLContextMultiKMS protected(kms: List[KeyManagementService], sparkSession: SparkSession) {
+
+  protected var dataKeyPlainText: String = ""
+  val kms = new scala.collection.mutable.list[KeyManagementService]
+  /**
+   * Load keys from a local file system.
+   * @param primaryKeyPath
+   * @param dataKeyPath
+   * @return
+   */
+  def loadKeys(primaryKeyPath: String, dataKeyPath: String, dataSourceName: String): this.type = {
+    dataKeyPlainText = kms.retrieveDataKeyPlainText(
+      new Path(primaryKeyPath).toString, new Path(dataKeyPath).toString,
+      sparkSession.sparkContext.hadoopConfiguration)
+    sparkSession.sparkContext.hadoopConfiguration.set(dataSourceName, dataKeyPlainText)
+    this
+  }
+
+  /**
+   * Read data files into RDD[String]
+   * @param path data file path
+   * @param minPartitions min partitions
+   * @param cryptoMode crypto mode, such as PLAIN_TEXT or AES_CBC_PKCS5PADDING
+   * @return
+   */
+  def textFile(path: String,
```

Comment:
As comment above, prefer to add new methods to PPMLContext instead of copying methods of PPMLContext here.